components/action-buttons/buyer-view-bidders-modal.tsx

```diff
@@ -0,0 +1,221 @@
+import BidderDetails from "./bidder-details";
+import { Controller, useForm } from "react-hook-form";
+import { zodResolver } from "@hookform/resolvers/zod";
+import { BidRequestSchema, BidRequestType } from "@/types/bidder";
+import { useContext, useEffect, useState } from "react";
+
+import Modal from "react-native-modal";
+import { ScrollView, View, ActivityIndicator } from "react-native";
+import { Button, Icon, Input, Text, useTheme } from "@ui-kitten/components";
+import { bidItem } from "@/firebase/functions";
+
+import NoBidders from "./no-bidders";
+import ConfirmBuyActionModal from "./confirm-buy-action-modal";
+import { BiddersModalContext } from "@/contexts/biddersModalContext";
+import { doc, onSnapshot } from "@react-native-firebase/firestore";
+import { itemsCollectionRef } from "@/constants/collections";
+
+function BuyerViewBiddersModal() {
+    const {
+        isBuyerViewModalVisible,
+        setIsBuyerViewModalVisible,
+        item,
+        bidders,
+    } = useContext(BiddersModalContext);
+    const itemId = item?.id ?? "";
+    const [isSteal, setIsSteal] = useState(false);
+    const theme = useTheme();
+    const [isLoading, setIsLoading] = useState(false);
+    const [isConfirmVisible, setIsConfirmVisible] = useState(false);
+    const [bidData, setBidData] = useState<BidRequestType | null>(null);
+
+    const {
+        handleSubmit,
+        formState: { errors },
+        control,
+        getValues,
+        reset,
+    } = useForm<BidRequestType>({
+        resolver: zodResolver(BidRequestSchema),
+        defaultValues: {
+            itemId: itemId,
+        },
+        disabled: isLoading,
+    });
+
+    function setConfirmModalParams(data: BidRequestType | null) {
+        setBidData(data);
+        setIsConfirmVisible(true);
+    }
+
+    async function handlePlaceBid(data: BidRequestType | null) {
+        setIsLoading(true);
+        try {
+            const response = await bidItem(data);
+            console.log(response);
```

Comment:
delete log, and here lang mag reset sa field, kay if error di nato ganahan mawala